autotest/pyscripts/test_gdal_retile.py

```diff
@@ -31,6 +31,7 @@
 import shutil
 import os
 
+import numpy as np
```

Comment:
Although strongly suggested, not all GDAL builds have numpy available.

You can instead add the following snippet in your new test method
```
    try:
        from osgeo import gdalnumeric
        gdalnumeric.zeros
        import numpy as np
    except (ImportError, AttributeError):
        pytest.skip()
```